File: ProductAPI/ProductAPI/Entities/Product.cs
using System.ComponentModel.DataAnnotations;

namespace ProductAPI.Entities
{
    public class Product
    {
        [Key]
        public int Id { get; set; }
        public required string ProductName { get; set; }
        public decimal Price { get; set; }
        public required string ImageUrl { get; set; }
        public required string ProductUrl { get; set; }

    }
}

File: client/model/product.ts
export interface Product {
  items: Item[];
  totalItems: number;
  pageNumber: number;
  pageSize: number;
  totalPages: number;
}

export interface Item {
  id: number;
  productName: string;
  price: number;
  imageUrl: string;
  productUrl: string;
}
